### PR TITLE
Return API error messages for non successful requests

### PR DESCRIPTION
**Pull request by sgalsaleh**

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -71,26 +71,26 @@ func (conn *Connection) rqliteApiCall(apiOp apiOperation, method string, request
...
 		trace("%s: ioutil.ReadAll() OK", conn.ID)
 
+		// Check that we've got a successful answer
+		if response.StatusCode != http.StatusOK {
+			trace("%s: got code %s", conn.ID, response.Status)
+			failureLog = append(failureLog, fmt.Sprintf("%s failed, got: %s, message: %s", url, response.Status, string(responseBody)))
+			response.Body.Close()
+			continue
+		}
+		response.Body.Close()
```
Comment (otoolep) [inline api.go:91]: Should we make this `defer response.Body.Close()` after `responseBody, err := ioutil.ReadAll(response.Body)` and remove the explicit calls to `response.Body.Close()` elsewhere?

**Comment by otoolep** [comment]
LGTM -- merge away (one comment).

File: api.go (hunk 1 above)
Comment (sgalsaleh) [reply api.go:91]: It wouldn't work the same since this is happening inside a for loop and the defer wouldn't run until the function returns.

Comment (otoolep) [reply api.go:91]: Ah yes.

**Pull request merged**